test/suites/test_push.py

```diff
@@ -0,0 +1,449 @@
+    srv = TarantoolServer()
+    srv.script = 'test/suites/box.lua'
+    srv.start()
+    srv.admin("box.schema.user.create('test', {password = 'test', " +
+              "if_not_exists = true})")
+    srv.admin("box.schema.user.grant('test', 'read,write,execute', 'universe')")
+
+    # Create server_function (for testing purposes).
+    srv.admin("""
+    function server_function()
+        x = {0,0}
+        while x[1] < 3 do
+            x[1] = x[1] + 1
+            box.session.push(x)
+        end
+        return x
+    end
+    """)
+
+    # Create tester space and on_replace trigger (for testing purposes).
+    srv.admin("""
+    box.schema.create_space(
+        'tester', {
+        format = {
+            {name = 'id', type = 'unsigned'},
+            {name = 'name', type = 'string'},
+        }
+    })
+    """)
+    srv.admin("""
+    box.space.tester:create_index(
+        'primary_index', {
+        parts = {
+            {field = 1, type = 'unsigned'},
+        }
+    })
+    """)
+    srv.admin("""
+    box.space.tester:create_index(
+        'primary_index', {
+        parts = {
+            {field = 1, type = 'unsigned'},
+        }
+    })
+    """)
+    srv.admin("""
+    function on_replace_callback()
+        x = {0,0}
+        while x[1] < 300 do
+            x[1] = x[1] + 100
+            box.session.push(x)
+        end
+    end
+    """)
+    srv.admin("""
+    box.space.tester:on_replace(
+        on_replace_callback
+    )
+    """)
+
+    return srv
+
+
+# Callback for on_push arg (for testing purposes).
+def push_callback(data, on_push_ctx=[]):
+                data[0][1] = data[0][1] + 1
+                on_push_ctx.append(data)
+
+
+@unittest.skipIf(sys.platform.startswith("win"),
+                 'Mesh tests on windows platform are not supported')
+class TestSuite_Push(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(self):
+        print(' PUSH '.center(70, '='), file=sys.stderr)
+        print('-' * 70, file=sys.stderr)
+
+    def setUp(self):
+        # Create server and extract helpful fields for tests.
+        self.srv = create_server()
+        self.host = self.srv.host
+        self.port = self.srv.args['primary']
+
+    def test_00_00_push_call_via_connect(self):
+        def assert_push_call(conn):
+            callback_res = []
+            with warnings.catch_warnings():
+                warnings.simplefilter('ignore', category=NetworkWarning)
+                resp = conn.call(
+                    'server_function',
+                    on_push=push_callback, 
+                    on_push_ctx=callback_res
+                )
+            self.assertEqual(callback_res, [[[1, 1]], [[2, 1]], [[3, 1]]])
+            self.assertEqual(resp.data[0], [3, 0])
+
+        conn = tarantool.connect(host=self.host, port=self.port, user='test', password='test')
+        # Check push from instance.
+        assert_push_call(conn)
+        conn.close()
+        # Stop instance.
+        self.srv.stop()
```

Comment:
It definitely should be in `teardown`

Reply:
Oh, it is already there. That's one more reason why it shouldn't be there too.

Reply:
Done